sys/shell/commands/sc_ipv6_nc.c

```diff
@@ -0,0 +1,137 @@
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @ingroup     sys_shell_commands.h
+ * @{
+ *
+ * @file
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#include <string.h>
+
+#include "kernel_types.h"
+#include "net/ng_ipv6/addr.h"
+#include "net/ng_ipv6/nc.h"
+#include "net/ng_netif.h"
+#include "thread.h"
+
+/* maximum length of L2 address */
+#define MAX_L2_ADDR_LEN (8U)
+
+static bool _is_iface(kernel_pid_t iface)
+{
+#ifdef MODULE_NG_NETIF
+    size_t numof;
+    kernel_pid_t *ifs = ng_netif_get(&numof);
+
+    for (size_t i = 0; i < numof; i++) {
+        if (ifs[i] == iface) {
+            return true;
+        }
+    }
+
+    return false;
+#else
+    return (thread_get(iface) != NULL);
+#endif
+}
+
+static size_t _parse_addr(char *str, uint8_t *addr)
+{
+    char *tok = strtok(str, ":");
+    size_t res = 0;
+
+    while (tok != NULL) {
+        if (res >= MAX_L2_ADDR_LEN) {
+            return 0;
+        }
+
+        uint16_t tmp = (uint16_t)strtol(tok, NULL, 16);
+
+        if (tmp <= 0xff) {
+            addr[res++] = (uint8_t)tmp;
+        }
+        else {
+            return 0;
+        }
+
+        tok = strtok(NULL, ":");
+    }
+
+    return res;
+}
+
+static void _ipv6_nc_add(kernel_pid_t iface, char *ipv6_addr_str,
+                         char *l2_addr_str)
+{
+    ng_ipv6_addr_t ipv6_addr;
+    uint8_t l2_addr[MAX_L2_ADDR_LEN];
+    size_t l2_addr_len;
+
+    if (ng_ipv6_addr_from_str(&ipv6_addr, ipv6_addr_str) == NULL) {
+        puts("error: unable to parse IPv6 address.");
+        return;
+    }
+
+    if ((l2_addr_len = _parse_addr(l2_addr_str, l2_addr)) == 0) {
+        puts("error: unable to parse link-layer address.");
+        return;
+    }
+
+    if (ng_ipv6_nc_add(iface, &ipv6_addr, l2_addr, l2_addr_len, 0) < 0) {
+        puts("error: unable to add address to neighbor cache.");
+        return;
+    }
+
+    printf("success: added IPv6 address %s to neighbor cache\n", ipv6_addr_str);
+}
+
+static void _ipv6_nc_del(char *ipv6_addr_str)
+{
+    ng_ipv6_addr_t ipv6_addr;
+
+    if (ng_ipv6_addr_from_str(&ipv6_addr, ipv6_addr_str) == NULL) {
+        puts("error: unable to parse IPv6 address.");
+        return;
+    }
+
+    ng_ipv6_nc_remove(KERNEL_PID_UNDEF, &ipv6_addr);
+    printf("success: removed %s from neighbor cache\n", ipv6_addr_str);
+}
+
+void _ipv6_nc_manage(int argc, char **argv)
+{
+    if (argc > 2) {
+        kernel_pid_t iface = (kernel_pid_t)atoi(argv[2]);
+
+        if ((argc > 4) && (strcmp("add", argv[1]) == 0)) {
+            if (!_is_iface(iface)) {
+                puts("error: invalid interface given.");
+                return;
+            }
+
+            _ipv6_nc_add(iface, argv[3], argv[4]);
+            return;
+        }
+
+        if (strcmp("del", argv[1]) == 0) {
+            _ipv6_nc_del(argv[2]);
+            return;
+        }
+    }
+
```

Comment:
I would vote for Output, since this information is targeted at the user of the shell commands... But I'm unsure how to do this in a concise way, so if it's impossible I'd be okay with it being in the code as well, as long as it's written down somewhere.